### PR TITLE
[WC-207] Write docs for new Maps widget

### PR DESCRIPTION
**Pull request by Keraito**
Similar to #3100:

This updates the Maps pluggable widget documentation that will only be supported on MX 9 and above. I kept the old widget documentation as well, since mx developers can still use these widgets with mx 7 or 8.

This needs to go through a technical review. So for you guys it might be best to delay proofread until after that has happened

### CODE REVIEW

**Comment by Adam-Dupaski** [comment]
Hi @Keraito the GitHub link on https://github.com/mendix/maps goes to an archived repo. Can that be updated? Because this means the *Documentation* tab in the Marketplace for the component is also pulling from the wrong repo.

**Comment by Adam-Dupaski** [comment]
@ConnorLand we'll see who has time for this once it's ready.

**Comment by Keraito** [comment]
> Hi @Keraito the GitHub link on https://github.com/mendix/maps goes to an archived repo. Can that be updated? Because this means the Documentation tab in the Marketplace for the component is also pulling from the wrong repo.

@Adam-Dupaski Will take a look 👍

File: content/appstore/widgets/maps.md [hunk 1]
```diff
@@ -6,7 +6,95 @@ tags: ["marketplace", "marketplace component", "widget", "maps", "google maps",
...
+To make the process of marking batches of locations easier, there's also the possibility to specify entire lists of markers at once.
+This can be done through the **General > Markers > Marker list** setting and requires a data source.
+All the usual Mendix data source settings apply here, as well as the same marker requirements and customizability as the individual markers. 
+
+### Controls
+
+Under the **Controls** properties tab, the following settings can be adjusted that are related to how the users interacts with the map:
+* **Drag**: When enabled, the map will move along when dragging the map.
+* **Scroll to zoom**: When enabled, the user can change the zoom level of the map by mouse scrolling.*
+* **Zoom**: When enabled, additional control buttons will be showed on the map
```
Comment (IIsaku) [inline content/appstore/widgets/maps.md:76]: ... will be shown

File: content/appstore/widgets/maps.md [hunk 2]
```diff
@@ -6,7 +6,95 @@ tags: ["marketplace", "marketplace component", "widget", "maps", "google maps",
...
+
+To make the process of marking batches of locations easier, there's also the possibility to specify entire lists of markers at once.
+This can be done through the **General > Markers > Marker list** setting and requires a data source.
+All the usual Mendix data source settings apply here, as well as the same marker requirements and customizability as the individual markers. 
+
+### Controls
+
+Under the **Controls** properties tab, the following settings can be adjusted that are related to how the users interacts with the map:
+* **Drag**: When enabled, the map will move along when dragging the map.
+* **Scroll to zoom**: When enabled, the user can change the zoom level of the map by mouse scrolling.*
```
Comment (IIsaku) [inline content/appstore/widgets/maps.md:75]: I think zooming will also work on mobile devices with pinch gestures.

File: content/appstore/widgets/maps.md [hunk 3]
```diff
@@ -6,7 +6,95 @@ tags: ["marketplace", "marketplace component", "widget", "maps", "google maps",
...
+* **Street view**: When enabled, the user will have an icon button available in the bottom right corner to view the map in Google's street view mode.
+* **Map type**: When enabled, the user will be provided control buttons in the top left corner to change the type of the map. Options available are **Map**, with or without a **Terrain** layer, and **Satellite**, with or without a **Labels** layer. By default this is set to **Map** without the **Terrain** layer.
+* **Full screen**: When enabled, the user will have an icon button available in the top right corner to view the map in full screen.
+
+_\* - When using the Google Maps provider, the **scroll to zoom** setting requires the **drag** setting to be enabled in order to work._
+
+### Dimensions
+
+Under the **Dimensions** properties tab, the following settings can be adjusted that are related to dimensional aspects of the map widgets:
+* **Width unit** and **Width**: The width of the widget in relation to the rest of the elements on the page. The available options for unit are _Percentage_ and _Pixels_. The width can then be set as an appropriate CSS value. These two properties need to be used together to work.
```
Comment (IIsaku) [inline content/appstore/widgets/maps.md:89]: What do you mean by "The width can then be set as an appropriate CSS value"?

File: content/appstore/widgets/maps.md [hunk 4]
```diff
@@ -6,7 +6,95 @@ tags: ["marketplace", "marketplace component", "widget", "maps", "google maps",
...
+* **Height unit** and **Height**: The height of the widget in relation to the rest of the elements on the page. The available options for unit are _Percentage of width_, _Pixels_, and _Percentage of parent_. The height can then be set as an appropriate CSS value. These two properties need to be used together to work.
+* **Zoom level**: The starting zoom level of the map. Options are: Automatic, World, Continent, City, Street, and Buildings.
```
Comment (IIsaku) [inline content/appstore/widgets/maps.md:91]: How does this setting relate to having all markers visible on the map?

**Review by IIsaku — COMMENTED** [review]
@Keraito Some remarks and questions for you, but overall looking really good. Well done! 👍

File: content/appstore/widgets/maps.md (hunk 3 above)
Comment (Keraito) [reply content/appstore/widgets/maps.md:89]: My train of thought went as follows: you need to use the **width unit** and **width** properties together. For unit, you can choose either _Percentage_ or _Pixels_. For the value (**Width**), you can then specify the appropriate value based on what you would do in CSS (100%, 200px)

File: content/appstore/widgets/maps.md (hunk 4 above)
Comment (Keraito) [reply content/appstore/widgets/maps.md:91]: Not quite sure, will have to check

File: content/appstore/widgets/maps.md (hunk 1 above)
Comment (Keraito) [reply content/appstore/widgets/maps.md:76]:
```suggestion
* **Zoom**: When enabled, additional control buttons will be shown on the map
```

File: content/appstore/widgets/maps.md (hunk 2 above)
Comment (Keraito) [reply content/appstore/widgets/maps.md:75]: will check

**Comment by Keraito** [comment]
> Hi @Keraito the GitHub link on https://github.com/mendix/maps goes to an archived repo. Can that be updated? Because this means the Documentation tab in the Marketplace for the component is also pulling from the wrong repo.

@Adam-Dupaski where exactly should I change this link?

File: content/appstore/widgets/maps.md (hunk 4 above)
Comment (Keraito) [reply content/appstore/widgets/maps.md:91]: So the "having all markers visible on the map" basically just moves the center of the map to the center of all markers. This setting then zooms in or out on that location. It's kind of odd, so I will specify it in the docs in a bit more detail

Comment (Keraito) [reply content/appstore/widgets/maps.md]: Fixed

File: content/appstore/widgets/maps.md (hunk 2 above)
Comment (Keraito) [reply content/appstore/widgets/maps.md:75]:
```suggestion
* **Scroll to zoom**: When enabled, the user can change the zoom level of the map by mouse scrolling on desktop or using pinch gestures on mobile.*
```

**Comment by Adam-Dupaski** [comment]
> 
> 
> > Hi @Keraito the GitHub link on https://github.com/mendix/maps goes to an archived repo. Can that be updated? Because this means the Documentation tab in the Marketplace for the component is also pulling from the wrong repo.
> 
> @Adam-Dupaski where exactly should I change this link?

@Keraito sorry, wrong like in the message. The GitHub links on the Marketplace page for the component https://marketplace.mendix.com/link/component/108261 goes to the archived repo. So the link to GitHub in the Marketplace needs to be fixed.

File: content/appstore/widgets/maps.md (hunk 3 above)
Comment (IIsaku) [reply content/appstore/widgets/maps.md:89]: Clear. Resolved.

File: content/appstore/widgets/maps.md (hunk 4 above)
Comment (IIsaku) [reply content/appstore/widgets/maps.md]: Nice adjustment! Resolved

**Review by IIsaku — APPROVED** [review]

**Comment by Adam-Dupaski** [comment]
How much more work is needed on this PR before review can be done?

**Comment by Keraito** [comment]
@Adam-Dupaski Isa reviewed this so now you guys can pick this up for language review. Meanwhile I will sort out how we can adjust the link for documentation that you mentioned before we launch this

**Review by Adam-Dupaski — APPROVED** [review]

**Comment by Keraito** [comment]
@Adam-Dupaski Please do not merge this for 9.0.5 yet, we found an issue that we want to fix before re-releasing the widget but won't make that it for the 9.0.5 release so we'll do it after. I'll let you know here when it's done.

**Comment by Keraito** [comment]
@Adam-Dupaski we fixed it and released the maps widget, so please merge this whenever you can 😄

**Pull request merged**